### PR TITLE
Refactor the default selection

### PR DESCRIPTION
**Pull request by haddadanas**
This PR improves the following on the code in the `default.py` in the selection:

- Refactor the code parts in the `default` selector into `particle_selection`, `mc_selection` and `selection_mc_weights` for better maintainability and convenience when creating or modifying a selector
- ~~Added a new empty selector, which creates all columns added in the default selector, but returns an all True mask~~

The results of the refactored selector was compared with results of the original one for the dataset `data_mu_e`.

### CODE REVIEW

File: hbt/selection/default.py [hunk 1]
```diff
@@ -78,51 +69,81 @@ def default(
     events, jet_results = self[jet_selection](events, trigger_results, lepton_results, **kwargs)
     results += jet_results
 
-    # mc-only functions
-    if self.dataset_inst.is_mc:
-        events = self[mc_weight](events, **kwargs)
+    # create process ids
+    events = self[process_ids](events, **kwargs)
```
Comment (nprouvost) [inline hbt/selection/default.py:73]: not a new selection, so this should probably be in the main Selectors (default and "empty"), instead of here

**Review by nprouvost — CHANGES_REQUESTED** [review]

File: hbt/selection/default.py [hunk 2]
```diff
@@ -169,5 +240,66 @@ def default(
...
+    },
+    produces={
+        trigger_selection, lepton_selection, jet_selection, mc_weight,
+        pdf_weights, murmuf_weights, pu_weight, btag_weights, process_ids, cutflow_features,
+        increment_stats,
+    },
+    sandbox=dev_sandbox("bash::$HBT_BASE/sandboxes/venv_columnar_tf.sh"),
+    exposed=True,
+)
+def empty(
```
[the remark this answers is not in the document]
Comment (haddadanas) [reply hbt/selection/default.py:259]: Back then I didn't know about the empty cf selector or it hasn't existed yet.
I assume this is then not needed and can be removed then or?

File: hbt/selection/default.py [hunk 3]
```diff
@@ -78,51 +69,81 @@ def default(
...
+    uses={
+        mc_weight, pdf_weights, murmuf_weights, pu_weight, btag_weights,
+    },
+    produces={
+        mc_weight, pdf_weights, murmuf_weights, pu_weight, btag_weights,
+    },
+    sandbox=dev_sandbox("bash::$HBT_BASE/sandboxes/venv_columnar_tf.sh"),
+    exposed=False,
+)
+def mc_selection(
```
[the remark this answers is not in the document]
Comment (haddadanas) [reply hbt/selection/default.py:88]: I understand, but wouldn't we have then a mix between the selection and production modules?

[the remark this answers is not in the document]
Comment (riga) [reply hbt/selection/default.py]: This wouldn't be a problem. In fact, that's actually desired :)

**Comment by riga** [comment]
Closed after private discussion.

**Pull request closed without merging**